ui/cypress/integration/e2e/userpreference_spec.js

```diff
@@ -0,0 +1,45 @@
+describe('User Preferences', () => {
+  describe('Analytics and Improvement Program', () => {
+    beforeEach(() => {
+      cy.route2('GET', '/api/user/stats').as('getUserStats')
+
+      cy.selectProviderNone()
+
+      cy.visit('/userpreference')
+      cy.get('.MuiFormLabel-root').should('have.text', 'Analytics and Improvement Program')
+      cy.wait('@getUserStats')
+    })
+
+    it('deactivates "Send Anonymous Usage Statistics"', () => {
+      cy.route2('POST', '/api/user/stats').as('postUserStats')
+
+      cy.get('fieldset > div > label:nth-child(1)').click()
+      cy.wait('@postUserStats')
+      cy.get('.MuiSnackbarContent-message').should('have.text', 'Sending anonymous usage statistics was disabled')
+    })
+
+    it('activates "Send Anonymous Usage Statistics"', () => {
+      cy.route2('POST', '/api/user/stats').as('postUserStats')
+
+      cy.get('fieldset > div > label:nth-child(1)').click()
+      cy.wait('@postUserStats')
+      cy.get('.MuiSnackbarContent-message').should('have.text', 'Sending anonymous usage statistics was enabled')
```

Comment:
@ramrodo looking good. Two thoughts:

1) The text used in these messages might be updated over time. Is there any other approach to validating the test passed or failed? If there is no less fragile way to design this test, then we'll just be sure to update these tests as we go.

2) We have overused (abused) snackbars in Meshery's current user experience. Hopefully, soon the UX will improve to reduce the noise here and start to show other visual indicators of a switch being set to on or off.

Reply:
Yes, there is another way to validate if that test pass. We can check if the switch button is disabled or if the CSS style changed. Let me update it and remove that message validation. It's challenging to maintain these successful messages as you mention.